src/__experimental__/components/tooltip/tooltip.spec.js

```diff
@@ -0,0 +1,338 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import 'jest-styled-components';
+import { mount } from 'enzyme';
+import Tooltip from '.';
+import { StyledTooltipInner, StyledTooltipPointer, StyledTooltipWrapper } from './tooltip.style';
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+import OptionsHelper from '../../../utils/helpers/options-helper';
+
+function render(props, renderer = TestRenderer.create) {
+  return renderer(<Tooltip { ...props }>Content</Tooltip>);
+}
+
+function renderInner(props) {
+  return TestRenderer.create(<StyledTooltipInner { ...props } />).toJSON();
+}
+
+function renderPointer(props) {
+  return TestRenderer.create(<StyledTooltipPointer { ...props } />).toJSON();
+}
+
+function renderWrapper(props) {
+  return TestRenderer.create(<StyledTooltipWrapper { ...props } />).toJSON();
+}
+
+describe('Tooltip', () => {
+  describe('default props', () => {
+    it('matches snapshot', () => {
+      const wrapper = render({}, mount);
+
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('visible', () => {
+    it('matches snapshot', () => {
+      const wrapper = render({ isVisible: true }).toJSON();
+
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('styles', () => {
+    describe('TooltipInner', () => {
+      it('default', () => {
+        assertStyleMatch(
+          {
+            backgroundColor: '#00141D',
+            color: '#FFFFFF',
+            display: 'inline-block',
+            fontWeight: '700',
+            padding: '10px 15px',
+            textAlign: 'center',
+            maxWidth: '300px',
+            wordBreak: 'normal',
+            whiteSpace: 'pre-wrap'
+          },
+          renderInner()
+        );
+      });
+
+      it('type="error"', () => {
+        assertStyleMatch(
+          { backgroundColor: '#C7384F' }, renderInner({ type: 'error' })
+        );
+      });
+    });
+
+    describe('TooltipWrapper', () => {
+      it('default', () => {
+        assertStyleMatch(
+          {
+            position: 'absolute',
+            width: '300px',
+            zIndex: '1003'
+          },
+          renderWrapper(),
+        );
+      });
+
+      OptionsHelper.positions.forEach((position) => {
+        it(`position === ${position}`, () => {
+          const align = ['top', 'bottom'].includes(position) ? 'center' : position;
+
+          assertStyleMatch(
+            { textAlign: align },
+            renderWrapper({ position })
+          );
+        });
+      });
```

Comment:
if I understand correctly - for position `top` and `bottom`, the text have to be centered? if yes, then it could be done like that:
```
describe.each(['top','bottom'])('for the position prop set to "%s"', (align) => {
  it('should have it's text aligned to "center"', () => {
```

Reply:
Refactored the specs to use describe.each (I've also refactored quite a few others to give better description / test names.

I _haven't_ separated top / bottom and left / right out for this specific spec out into two though, as I could still achieve the same thing with the ternary, and less duplication of code